docs/site/exposing-graphql-apis.md

```diff
@@ -0,0 +1,131 @@
+---
+lang: en
+title: 'Exposing GraphQL APIs'
+keywords: LoopBack 4.0, LoopBack 4
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/exposing-graphql-apis.html
+---
+
+## Overview
+
+The [OASGraph module](https://www.npmjs.com/package/oasgraph) creates a GraphQL
+wrapper for existing REST APIs which are described by the OpenAPI specification.
+This tutorial shows how to expose GraphQL APIs in an existing LoopBack
+application.
+
+### Prerequisite
+
+Make sure you have a running LoopBack 4 application. In this tutorial, we'll use
+the `todo` example. You can create this application by running the command
+below:
+
+```sh
+lb4 example todo
+```
+
+### Customize the Server URL of the REST Server
+
+OASGraph expects the provided OpenAPI specification to have the `servers` url to
+be the server hosting the REST endpoints. We're going to modify the constructor
+of `TodoListApplication` in `src/application.ts` to include this information.
+
+```ts
+constructor(options: ApplicationConfig = {}) {
+    // Add this snippet at the beginning of the constructor
+    options = Object.assign(
+      {},
+      {
+        rest: {
+          openApiSpec: {
+            servers: [{url: 'http://localhost:3000'}],
+          },
+        },
+      },
+    );
+    // end of the code snippet
+    super(options);
+    ...
+}
+```
+
+### Install OASGraph and Required Dependencies
+
+From your LoopBack application, run the following command to install OASGraph
+and the required dependencies:
+
+```sh
+npm i --save oasgraph express-graphql
+```
+
+### Start the GraphQL Server
+
+Make sure your LoopBack application is running by going to
+`http://localhost:3000/openapi.json`. If not, you can start it by running the
+`npm start` command.
+
+Now we will use the oasgraph CLI to set up a GraphQL HTTP Server backed by
+express on port 3001. Specifying the OpenAPI spec generated by the
+todo-application as the parameter, start up the server by running the following
+command:
+
+```sh
+node_modules/.bin/oasgraph http://localhost:3000/openapi.json
```

Comment:
This can be simplified as `npx oasgraph http://localhost:3000/openapi.json`

Reply:
Just need to mention the user that `npx`  comes bundled with npm version >5.2 . it works fine, I have just tested, didn't know about `npx` 😄

Reply:
Node.js 8.9.0, the minimal Node.js version required by LoopBack 4, comes with npm version 5.5.1, which includes `npx`. IMO, there is no need to explain how to run oasgraph on npm version older than 5.2.

Having said that, I think most LoopBack users are probably not familiar with npx yet, so we should give them a pointer on where to learn more.

For example, add the following paragraph after the code snippet showing how to invoke `npx`:

```markdown
_Haven't heard about `npx` yet? It's a cool helper provided by `npm` 
and available out of the box since Node.js 8.x. Learn more in their announcement blog post:
[Introducing npx: an npm package runner](https://medium.com/@maybekatz/introducing-npx-an-npm-package-runner-55f7d4bd282b))_
```